mountpoint-s3/src/inode.rs

```diff
@@ -663,13 +726,13 @@ impl WriteHandle {
         };
 
         // Acquire locks on ancestors in descending order to avoid deadlocks.
-        let mut ancestors_states: Vec<_> = ancestors
+        let ancestors_states: Result<Vec<_>, InodeError> = ancestors
             .iter()
             .rev()
-            .map(|inode| inode.inner.sync.write().unwrap())
+            .map(|inode| inode.get_mut_inode_state())
             .collect();
```

Comment:
Almost there. But we want to return an error here, rather than after acquiring the inode lock.

Maybe this is a case for the turbofish?

```
let mut ancestors_states = ancestors
    .iter()
    .rev()
    .map(|inode| inode.get_mut_inode_state())
    .collect::<Result<Vec<_>, _>>()?;
```

Reply:
Thanks. It was good to learn about turbofish. Doing the change.